api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+    later moment this will be configurable and not be inherent part of this API).
+
+    * Topic API (low level).
+
+    Other aspects of the Event Bus are at this moment to be defined and otherwise specificied, not included
+    in this version of this specification.
+
+    Notable omissions here are:
+    * The contract between Nakadi and clients in the context of the high-level API (i.e. the process for
+    clients to establish subscriptions).
+
+    * Submission of events to topics in batched manner.
+
+    * Security scopes (OAuth) for the different operations
+
+    * Validation and Enrichment Strategies available in Nakadi are out of this document an can be consulted
+    in an external document (TBD link to document).
+
+
+  version: '0.5'
+  contact:
+    name: Team Aruha @ Zalando
+    email: team-aruha+nakadi-maintainers@zalando.de
+schemes:
+  - https
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  oauth2:
+    type: oauth2
+    flow: implicit
+    authorizationUrl: 'https://auth.example.com/oauth2/tokeninfo'
+
+paths:
+  /metrics:
+    get:
+      tags:
+        - monitoring
+      summary: Get monitoring metrics
+      responses:
+        '200':
+          description: Ok
+          schema:
+            $ref: '#/definitions/Metrics'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Not available
+          schema:
+            $ref: '#/definitions/Problem'
+  /event-types:
+    get:
+      tags:
+        - schema-registry-api
+      description: Returns a list of all registered `EventType`s
+      parameters:
+        - name: X-Flow-Id
+          in: header
+          description: |
+            The flow id of the request, which is written into the logs and passed to called services. Helpful
+            for operational troubleshooting and log analysis.
+          type: string
+          format: flow-id
+      responses:
+        '200':
+          description: Ok
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/EventType'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+
+    post:
+      tags:
+        - schema-registry-api
+      description: |
+        Creates a new `EventType`.
+
+        **Implementation note:** The creation of an EventType implicitly creates the structures in the
+        backing messaging implementation needed for the reception and persistency of the Events. Considering
+        that at this time only Kafka is used, this corresponds to the creation of a Topic. If so desired,
```

Comment:
@regispl Thank you for the comment. You are right, that 1:1 relationship of EventType to an ordering tool like Topic could be not sufficient.

Our considersations here are mostly related to the following issues:
* the only reason to push more then one event-type into the topic is the need to retain ordering. That also means, that these event-types should have the same ordering key (partitioning key). We even considered the possibility of mapping partitioning keys to topics
* applications should be made responsible of synchronisation of the the flow in such a unified topic and this is much more difficult to achieve then for one event-type. Especially if entities are produced by separate applications (that do not have the means of strong synchronisation usually), so the more correct way would have been not the mapping (`partitioning key` -> `topic`), but (`application`,`partitioning key` -> `topic`) as a partitioning tool, that reflects the real ordering guarantee for the consumer. But such a construct is much more difficult to understand and keep track of.

As @jkliff already mentioned, we definitely need a correlation engine (probably on the consumer side), that would reorder events coming from different topics. But such an engine is out of scope for us for the current milestone.